src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,206 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (isEmpty()) {
+            linkEmptyList(value);
+            return;
+        }
+        Node<T> newNode = new Node<>(tail, value);
+        newNode.prev.next = newNode;
+        tail = newNode;
+        newNode.prev = tail.prev;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == 0 && size == 0) {
+            linkEmptyList(value);
+            return;
+        }
+        if (index == 0) {
+            linkHead(value);
+            return;
+        }
+        if (index == size) {
+            linkTail(value);
+            return;
+        }
+        checkIndexOutOfBounds(index);
+        Node<T> current = getNode(index);
+        Node<T> newNode = new Node<>(current.prev,value,current);
+        current.prev.next = newNode;
+        current.prev = newNode;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        if (list.size() == 0) {
+            return;
+        }
+        for (T element:list) {
```

Comment:
Check is unnecessary, there won't be any iteration if size == 0
```suggestion
        for (T element : list) {
```
make sure to autoformat using IDE the files you are committing